presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java

```diff
@@ -995,19 +996,17 @@ private String validateAndNormalizeAvroSchemaUrl(String url, HdfsContext context
         }
     }
 
-    private Path getExternalPath(HdfsContext context, String location)
+    private void checkExternalPath(HdfsContext context, Path path)
     {
         try {
-            Path path = new Path(location);
             if (!isS3FileSystem(context, hdfsEnvironment, path)) {
                 if (!hdfsEnvironment.getFileSystem(context, path).isDirectory(path)) {
-                    throw new PrestoException(INVALID_TABLE_PROPERTY, "External location must be a directory: " + location);
+                    throw new PrestoException(INVALID_TABLE_PROPERTY, "External location must be a directory: " + path);
                 }
             }
-            return path;
         }
-        catch (IllegalArgumentException | IOException e) {
-            throw new PrestoException(INVALID_TABLE_PROPERTY, "External location is not a valid file system URI: " + location, e);
+        catch (IOException e) {
+            throw new PrestoException(INVALID_TABLE_PROPERTY, "External location is not a valid file system URI: " + path, e);
```

Comment:
This was supposed to guard the Path ctor (`Path path = new Path(location);`)
see my comment in the new place